models/migrations/v226.go

```diff
@@ -0,0 +1,210 @@
+		IsSystemWebhook    bool
+		URL                string `xorm:"url TEXT"`
+		HTTPMethod         string `xorm:"http_method"`
+		ContentType        webhook.HookContentType
+		Secret             string `xorm:"TEXT"`
+		Events             string `xorm:"TEXT"`
+		*webhook.HookEvent `xorm:"-"`
+		IsActive           bool               `xorm:"INDEX"`
+		Type               webhook.HookType   `xorm:"VARCHAR(16) 'type'"`
+		Meta               string             `xorm:"TEXT"` // store hook-specific attributes
+		LastStatus         webhook.HookStatus // Last delivery status
+
+		// HeaderAuthorizationEncrypted should be accessed using HeaderAuthorization() and SetHeaderAuthorization()
+		HeaderAuthorizationEncrypted string `xorm:"TEXT"`
+
+		CreatedUnix timeutil.TimeStamp `xorm:"INDEX created"`
+		UpdatedUnix timeutil.TimeStamp `xorm:"INDEX updated"`
+	}
+	err := x.Sync2(new(Webhook))
+	if err != nil {
+		return err
+	}
+
+	// Migrate the matrix webhooks
+
+	type MatrixMeta struct {
+		HomeserverURL string `json:"homeserver_url"`
+		Room          string `json:"room_id"`
+		MessageType   int    `json:"message_type"`
+	}
+	type MatrixMetaWithAccessToken struct {
+		MatrixMeta
+		AccessToken string `json:"access_token"`
+	}
+
+	var start int
+	hooks := make([]*Webhook, 0, 50)
+	for {
+		err := x.Where("type=?", "matrix").OrderBy("id").Limit(50, start).Find(&hooks)
+		if err != nil {
+			return err
+		}
+		if len(hooks) == 0 {
+			break
+		}
+
+		err = func() error {
+			sess := x.NewSession()
+			defer sess.Close()
+			if err := sess.Begin(); err != nil {
+				return fmt.Errorf("unable to allow start session. Error: %w", err)
+			}
+			for _, hook := range hooks {
+				// retrieve token from meta
+				var withToken MatrixMetaWithAccessToken
+				err := json.Unmarshal([]byte(hook.Meta), &withToken)
+				if err != nil {
+					return fmt.Errorf("unable to unmarshal matrix meta for webhook[%d]: %w", hook.ID, err)
+				}
+				if withToken.AccessToken == "" {
+					continue
+				}
+
+				// encrypt token
+				authorization := "Bearer " + withToken.AccessToken
+				hook.HeaderAuthorizationEncrypted, err = secret.EncryptSecret(setting.SecretKey, authorization)
+				if err != nil {
+					return fmt.Errorf("unable to encrypt access token for webhook[%d]: %w", hook.ID, err)
+				}
+
+				// remove token from meta
+				withoutToken, err := json.Marshal(withToken.MatrixMeta)
+				if err != nil {
+					return fmt.Errorf("unable to marshal matrix meta for webhook[%d]: %w", hook.ID, err)
+				}
+				hook.Meta = string(withoutToken)
+
+				// save in database
+				count, err := sess.ID(hook.ID).Cols("meta", "header_authorization_encrypted").Update(hook)
+				if count != 1 || err != nil {
+					return fmt.Errorf("unable to update header_authorization_encrypted for webhook[%d]: %d,%w", hook.ID, count, err)
+				}
+			}
+			return sess.Commit()
+		}()
+		if err != nil {
+			return err
+		}
+
+		if len(hooks) < 50 {
+			break
+		}
+		start += 50
+		hooks = hooks[:0]
+	}
```

Comment:
I created a generic `batchProcess` function.

Since this logic has been used in a couple of migrations (112, 210, 221 and this one), maybe it could make sens to add this method to xorm?